prime-router/src/main/kotlin/azure/ApiSearch.kt

```diff
@@ -183,21 +190,34 @@ abstract class ApiSearch<PojoType, RecordType : Record, ApiFilterType : ApiFilte
      * @return the list of the records parsed into the [PojoType]
      *
      */
-    open fun <T : Record> fetchResults(dslContext: DSLContext, select: SelectJoinStep<T>): ApiSearchResult<PojoType> {
-        val totalCount = dslContext.fetchCount(select)
-        val filteredCount = dslContext.fetchCount(
-            select
-                .where(getWhereClause())
-        )
-        val results = dslContext.fetch(
-            select
-                .where(getWhereClause())
-                .orderBy(getSortClause(), getPrimarySortClause())
-                .limit(limit)
-                .offset(getOffset())
-        ).into(recordClass)
-        return ApiSearchResult(totalCount, filteredCount, results)
-    }
+    open fun <T : Record> fetchResults(dslContext: DSLContext, select: SelectJoinStep<T>): ApiSearchResult<PojoType> =
+        runBlocking {
```

Comment:
There is no reason that these each need to run sequentially, so issuing them in three coroutines so that they can run in parallel